includes/class-search-categories-metabox.php

```diff
@@ -0,0 +1,326 @@
+            $selected_categories = array();
+        }
+
+        if( isset( $search_query ) && ! is_null( $search_query ) ) {
+            $categories = $this->_get_taxonomy_categories( $search_query );
+        } else {
+            $categories = $this->_get_taxonomy_categories();
+        }
+
+        $result['category_list'] = '';
+        $result['selected_categories'] = $selected_categories;
+
+        if( ! empty( $categories ) ) {
+            ob_start();
+            foreach ($categories as $category) {
+                $is_user_selected = in_array( $category->term_id, $selected_categories );
+                ?>
+
+                <li id='<?php echo $category->term_id; ?>'>
+                    <label class="selecttit"><input id='in-category-<?php echo $category->term_id; ?>' class="scm-category-list-item" value=<?php echo $category->term_id; ?> name="post-category[]" type="checkbox" <?php if( $is_user_selected ) { echo 'checked="checked"'; } ?>> <?php echo $category->name; ?>
+                </li>
+
+                <?php
+            }
+            $result['category_list'] = ob_get_clean();
+        }
+
+        //echo json_encode( 'Daniel' );
+        wp_send_json( $result );
+        wp_die();
+    }
+
+    /**
+     * Loads the Javascripts requried for the Admin section.
+     *
+     * @since 0.1.0
+     */
+    public function load_admin_scripts() {
+
+        global $post;
+
+        wp_enqueue_script( 'admin_post_editor_script', plugin_dir_url( $this->plugin_file ) . 'assets/js/admin-script.js', array( 'jquery' ), $this->plugin_version );
+
+        wp_localize_script( 'admin_post_editor_script', 'ajax_object', array(
+            'ajax_url'   => admin_url( 'admin-ajax.php' ),
+            'ajax_nonce' => wp_create_nonce( "search-categories-metabox" ),
+            'post_id'    => $post->ID,
+        ) );
+
+    }
+
+    /**
+     * Loads the CSS required for the Admin section.
+     *
+     * @since 0.1.0
+     */
+    public function load_admin_styles() {
+
+        wp_enqueue_style( 'scm_admin_post_editor_css', plugin_dir_url( $this->plugin_file ) . 'assets/css/admin-style.css', false, $this->plugin_version );
+
+    }
+
+    /**
+     * Renders the plugin's settings page in the Dashboard.
+     *
+     * @since 0.1.0
+     */
+    public function render_search_categories_metabox_page() {
+
+        ?>
+        <div class="wrap">
+
+            <h1><?php echo esc_html( get_admin_page_title() ); ?></h1>
+
+            <form action='<?php echo esc_html( $_SERVER['REQUEST_URI'] ); ?>' method='POST'>
+                <?php
+                submit_button( 'Save Settings' );
+                ?>
+            </form>
+
+        </div>
+        <?php
+
+    }
+
+    /**
+     * Adds the options page in the Admin Dashboard.
+     *
+     * @since 0.1.0
+     */
+    public function add_admin_menu() {
+
+        add_options_page(
+            __( 'Search Categories Metabox', Search_Categories_Metabox::TEXT_DOMAIN ),
```

Comment:
Text domain is parsed by static code tools. So it should always be hardcoded as a string and not stored as a variable. More details at https://markjaquith.wordpress.com/2011/10/06/translating-wordpress-plugins-and-themes-dont-get-clever/